### ISSUE TITLE
Markdown publish FileRegexSubst regex example

### ISSUE THREAD

**Issue by tomcoonen** [vaab/gitchangelog#80]
@vaab Is there a regex example available for publishing with Markdown?

**Comment by vaab** [comment]
Hi, thanks for your interest in ``gitchangelog`` ! Are you using the provided mustache markdown template ? If no, can you provide an example of the 20 first lines of your changelog in markdown format ?
The regex won't be this complex to create.

**Comment by tomcoonen** [comment]
@vaab I'm using the provided default template, looks fine!

**Comment by vaab** [comment]
Ok, I won't have the time right now to do it. Maybe tomorrow.

**Comment by vaab** [comment]
Here you are:
```
OUTPUT_FILE = "CHANGELOG.rst"
INSERT_POINT_REGEX = r'''(?isxu)
^
(
  \s*\#\s+Changelog\s*(\n|\r\n|\r)        ## ``Changelog`` line
)

(                     ## Match all between changelog and release rev
    (
      (?!
         (?<=(\n|\r))                ## look back for newline
         \#\#\s+%(rev)s                     ## revision
         \s+
         \([0-9]+-[0-9]{2}-[0-9]{2}\)(\n|\r\n|\r)   ## date
      )
      .
    )*
)

(?P<tail>\#\#\s+(?P<rev>%(rev)s))
''' % {'rev': r"[0-9]+\.[0-9]+(\.[0-9]+)?"}

revs = [
    Caret(FileFirstRegexMatch(OUTPUT_FILE, INSERT_POINT_REGEX)),
    "HEAD"
]

publish = FileRegexSubst(OUTPUT_FILE, INSERT_POINT_REGEX, r"\1\o\n\g<tail>")
 ```
Please note that I had to release an updated markdown mustache template to support partial gitchangelog generation.
This info is now part of the ``README.rst`` also. I propably need to find a better way to provide templates and these info together...

**Comment by vaab** [comment]
So this is released in 3.0.2

**Comment by tomcoonen** [comment]
Great job, works like a charm!
